### ISSUE TITLE
Make Chimera update script more resilient against local schema changes

### ISSUE THREAD

**Issue by jstarek** [dCache/dcache#2903]
RT Ticket #9057 mentions database errors in the face of custom local schema changes. Should we introduce a message suggesting approaches to deal with such a situation into the script? In the reported case, the admin was able to update after undoing the schema changes and reapplied them afterwards.

**Comment by gbehrmann** [comment]
For what it is worth, the release notes already state that this (undoing local modifications) has to be done before upgrading.

**Comment by paulmillar** [comment]
No.  It's almost always wrong for an application to suggest how to fix a problem.  It is rare that an application has sufficient information to make such suggestions reliable: what works for one deployment doesn't necessarily work elsewhere, and doing the "wrong" thing may make things worse.

Instead, the error should describe the problem succinctly and accurately.  Doing this well should make the solution "obvious" to the admin.

With [RT #9057](http://rt.dcache.org/Ticket/Display.html?id=9057), the immediate problem is that the code logs a stack-trace: a stack-trace always means there's a bug in dCache.

Beyond this, we could improve the error message by detecting if the blocking object is part of the standard dCache; e.g.,

    non-dCache function filescacheinfo(character varying) depends on type t_locationinfo2

becomes

    non-dCache function filescacheinfo(character varying) depends on type t_locationinfo2

It's difficult to say (prima facie) how difficult such a change would be.

**Comment by paulmillar** [comment]
Another approach (somewhat similar to git) is to describe database as either "clean" or "dirty" depending on whether it has local modifications.  This would also give a clue (us, at least) to what might be causing the problem.

Both suggestions require the ability to detect non-standard elements within a database.  I believe Liquibase doesn't support this (currently), so we'd need some other mechanism.

**Comment by gbehrmann** [comment]
Correct, liquibase has no idea what the schema should be, only how to create it. Creating a second representation describing the intended state (rather than how to create it) would duplicate that information (since the intended state is the end result of the steps to create it - there is just no way to know that state without actually executing the changesets on the target RDBMS).

**Comment by kofemann** [comment]
Custom updates are not supported. Works as expected. Closing.
